lib/iris/_merge.py

```diff
@@ -1094,6 +1096,9 @@ def __init__(self, cube):
         self._hints = ['time', 'forecast_reference_time', 'forecast_period',
                        'model_level_number']
 
+        # The proto-cube source.
+        self._source = cube
+
```

Comment:
@dkillick We need to cache the source cube of the proto-cube, so that later on we can rebuild the cube signature in the event of `fill_value` promotion ...